sunpy/instr/aia/response.py

```diff
@@ -0,0 +1,308 @@
+        for c in channel_list:
+            index = channel_list.index(c)
+            self._channel_info[c] = {property: data_table[property][index]
+                                     for property in data_table.columns.keys()}
+
+    def calculate_effective_area(self, channel, **kwargs):
+        """
+        Calculate the effective area for a given channel of the instrument.
+
+        According to [1]_, the effective area is given by,
+
+        .. math::
+            A_{eff}(\lambda,t) = A_{geo}R_P(\lambda)R_S(\lambda)T_E(\lambda)
+            T_F(\lambda)D(\lambda,t)Q(\lambda),
+
+        in units of :math:`\mathrm{cm}^2`, where,
+
+        - :math:`A_{geo}`: geometrical collecting area
+        - :math:`R_P`, :math:`R_S`: reflectances of primary and secondary mirrors, respectively
+        - :math:`T_E`, :math:`T_F`: transmission efficiency of the entrance and focal-plane filters, respectively
+        - :math:`D`: correction to account for time-dependent contamination and deterioration
+        - :math:`Q`: quantum efficiency of the CCD
+
+        The effective area contains information about the efficiency of the
+        telescope optics and its sensitivity as a function of wavelength. All of the telescope
+        properties are read from the AIA instrument files available in SolarSoft.
+
+        Parameters
+        ----------
+        channel : `int`
+            wavelength center of the channel
+
+        Returns
+        -------
+        effective_area : array-like
+            effective area of the instrument over the entire wavelength range
+        """
+        include_crosstalk = kwargs.get('include_crosstalk', False)
+        reflectance = (self._channel_info[channel]['primary_mirror_reflectance']
+                       * self._channel_info[channel]['secondary_mirror_reflectance'])
+        transmission_efficiency = (self._channel_info[channel]['focal_plane_filter_efficiency']
+                                   * self._channel_info[channel]['entrance_filter_efficiency'])
+        effective_area = (self._channel_info[channel]['geometric_area_ccd']
+                          * reflectance*transmission_efficiency
+                          * self._channel_info[channel]['ccd_contamination']
+                          * self._channel_info[channel]['quantum_efficiency_ccd'])
+        if include_crosstalk:
+            effective_area += self.calculate_crosstalk(channel)
+
+        return effective_area
+
+    def calculate_crosstalk(self, channel):
+        """
+        Calculate effects due to channel crosstalk.
+
+        The focal-plane filters on Telescopes 1 (131 and 335) and 4 (94 and 304)
+        do not perfectly reject light from the opposite channels. This function
+        implements a correction for the given channel, based on the method in
+        SSW, specifically that in `sdo/aia/idl/response/aia_bp_blend_channels.pro`.
+        See section 2.2.1 and Figure 10 of [1]_ for more details.
```

Comment:
Reference not in docstring?

Am I missing something here, is it picking this up from somewhere else?